dali/pipeline/operators/reader/loader/lmdb.h

```diff
@@ -122,13 +115,25 @@ class LMDBReader : public Loader<CPUBackend, Tensor<CPUBackend>> {
   }
 
  private:
+  void Reset() override {
+    // work out how many entries to move forward to handle sharding
+    current_index_ = start_index(shard_id_, num_shards_, Size());
+    bool ok = lmdb::SeekLMDB(mdb_cursor_, MDB_FIRST, &key_, &value_);
+    DALI_ENFORCE(ok, "lmdb::SeekLMDB failed");
```

Comment:
Slighlty out of the scope of this PR but I think this error should be more explicit. Something like:
```suggestion
    DALI_ENFORCE(ok, "lmdb::SeekLMDB failed on MDB_FIRST with arguments key:"
                                 + to_string(key_) + " value:"  to_string(value_));
```